src/transit.js

```diff
@@ -341,19 +341,20 @@ class Transit {
 			if (payload.sender === this.nodeID) {
 				// Detect nodeID conflict
 				if (cmd === P.PACKET_INFO && payload.instanceID !== this.instanceID) {
-					return this.broker.fatal(
+					this.broker.fatal(
 						"ServiceBroker has detected a nodeID conflict, use unique nodeIDs. ServiceBroker stopped."
 					);
+					return this.Promise.resolve(false);
 				}
 
 				// Skip own packets (if only built-in balancer disabled)
 				if (cmd !== P.PACKET_EVENT && cmd !== P.PACKET_REQUEST && cmd !== P.PACKET_RESPONSE)
-					return;
+					return this.Promise.resolve(false);
 			}
 
 			// Request
 			if (cmd === P.PACKET_REQUEST) {
-				return this.requestHandler(payload);
+				return this.requestHandler(payload).then(() => true);
 			}
 
 			// Response
```

Comment:
For response handler no need the `return`?

Reply:
This was to ensure the return type for `messageHandler` is consistently `Promise<boolean>`.

This way the transporter can choose what it does according to whether the message was handled successfully. i.e. true or false.
If the `requestHandler` throws an error the catch will ensure false is returned.

Reply:
Great, thanks.